core/src/services/oss/core.rs

```diff
@@ -62,30 +64,41 @@ impl Debug for OssCore {
 }
 
 impl OssCore {
-    async fn load_credential(&self) -> Result<AliyunCredential> {
+    async fn load_credential(&self) -> Result<Option<AliyunCredential>> {
+        if self.allow_anonymous {
```

Comment:
Oh, the logic is different. Please return `Ok(None)` after `load`.

Reply:
By the way, we should always allow anonymous, so this field can be removed from `OssBuilder`.

Reply:
If deleted, then this prop is always in effect? A little confused.

Reply:
> If deleted, then this prop is always in effect

Yes, user don't need to set this.

Reply:
> And where can the core get this prop from?

It's not a setting that should be exposed to users. OpenDAL will always allow anonymous access if `ak`/`sk` not set.

Reply:
Ah that is what I did before....

Reply:
Sorry, my mistake.